src/services/AdcService/AdcEnums.ts

```diff
@@ -0,0 +1,13 @@
+import path from "path";
+import * as protobuf from 'protobufjs'
+import { adc1DataRate, adc2DataRate, analogIn, convMode, filterMode } from "./AdcTypes";
+
+const protoPckgPath = path.join(require.resolve('@edgepi-cloud/rpc-protobuf'),'..');
+
+const root = protobuf.loadSync(path.join(protoPckgPath, 'adc.proto'))
+
+export const AnalogIn = root.lookupEnum('AnalogIn').values as unknown as analogIn
+export const ADC1DataRate  = root.lookupEnum('ADC1DataRate').values as unknown as adc1DataRate
+export const ADC2DataRate = root.lookupEnum('ADC2DataRate').values as unknown as adc2DataRate
+export const FilterMode = root.lookupEnum('FilterMode').values as unknown as filterMode
+export const ConvMode = root.lookupEnum('ConvMode').values as unknown as convMode
```

Comment:
I feel like this is not a good way of type casting this. Could you explain what is happening here?

Reply:
The object returned from root.blah.values is [k:string]:number but I want it to specifically have properties AIN1...AIN8 (for analogIn for example). Since typescript doesn't know that it complains. This is a workaround, though I agree not the best..

However what I can do differently is have the analogIn type extend the object type that typescript expects and then no complaints and users will have properties AIN1..AIN8 autocomplete. But in that way if you enter another property that doesn't exist, for example analogIn.AIN12 there will be no complaints from typescript which is sort of annoying.

Reply:
Never mind about this. But maybe we should add a comment here for someone to look into this.